assets/js/util/index.js

```diff
@@ -590,30 +591,50 @@ export const findTagInHtmlContent = ( html, module ) => {
  * while requesting list of accounts.
  *
  * @param {string} module Module slug.
+ *
+ * @param {string|null} The tag id if found, otherwise null.
  */
 export const getExistingTag = async ( module ) => {
-	try {
-		let tagFound = data.getCache( module + '::existingTag', 300 );
-
-		if ( 'undefined' === typeof tagFound ) {
-			const html = await fetch( `${ googlesitekit.admin.homeURL }?tagverify=1&timestamp=${ Date.now() }` ).then( ( res ) => {
-				return res.text();
-			} );
-
-			tagFound = findTagInHtmlContent( html, module );
-			if ( ! tagFound ) {
-				tagFound = '';
+	const CACHE_KEY = `${ module }::existingTag`;
+	const { homeURL, ampMode } = googlesitekit.admin;
+
+	let tagFound = data.getCache( CACHE_KEY, 300 );
+
+	if ( ! tagFound ) {
```

Comment:
Shouldn't this become `'undefined' === typeof tagFound`? We might have misunderstood each other in the Slack convo, falsy tag checks should also be cached. Checking for undefined here ensures that if there's no cache, we re-check.

Reply:
Ok, gotcha. Yeah, I understood it as not found should not be cached. Is there a risk of this being queried too often?

Reply:
I think we shouldn't determine cache or not cache based on whether a result is positive or not. Only if an actual error happened, nothing should be cached.

There are some improvements to make regarding less aggressive caching (for which I'll open an issue for shortly), but I think we should be consistent for where we are now.